src/lang/java/class_def.rs

```diff
@@ -0,0 +1,168 @@
+    let mut constructors = vec![];
+    let mut methods = vec![];
+    let mut modifier = Modifier::new();
+
+    // Find bounds
+    let start;
+    let end;
+    if let Some(span) = ast.span {
+        start = span.0 as i32;
+        end = span.1 as i32;
+    } else {
+        start = 0;
+        end = 0;
+    }
+
+    // Generate the implementation data
+    for member in ast.children.iter() {
+        match &*member.r#type {
+            "modifiers" => {
+                modifier = parse_modifiers(member, &*component.path, &*component.package_name)
+            }
+            "class_body" | "interface_body" | "enum_body" | "annotation_body" => {
+                parse_class_body(
+                    member,
+                    &component,
+                    &mut constructors,
+                    &mut methods,
+                    &mut fields,
+                );
+            }
+            unknown_type => log_unknown_tag(unknown_type, "class"),
+        };
+    }
+
+    // Assemble the return value from what was extracted from the tree
+    Some(ClassOrInterfaceComponent {
+        component: ContainerComponent {
+            component,
+            accessor: modifier.accessor,
+            stereotype: stereotype,
+            methods,
+            container_name: instance_name,
+            line_count: end - start + 1,
+        },
+        declaration_type,
+        annotations: modifier.annotations,
+        constructors: constructors,
+        field_components: fields,
+    })
+}
+
+/// Parses the members of a class to find members, methods, blocks, etc.
+fn parse_class_body(
+    ast: &AST,
+    component: &ComponentInfo,
+    constructors: &mut Vec<MethodComponent>,
+    methods: &mut Vec<MethodComponent>,
+    fields: &mut Vec<FieldComponent>,
+) {
+    // Traverse body
+    for member in ast.children.iter() {
+        match &*member.r#type {
+            "constructor_declaration" | "static_initializer" => {
+                constructors.push(parse_method(member, component))
+            }
+            "method_declaration" => methods.push(parse_method(member, component)),
+            "field_declaration" => fields.append(&mut parse_field(member, component)),
+            "class_declaration"
+            | "interface_declaration"
+            | "enum_declaration"
+            | "annotation_declaration" => { /* None, since these were extracted + handled elsewhere */
+            }
+            unknown => log_unknown_tag(unknown, "class body"),
+        }
+    }
+}
+
+/// Parses a single field in a class
+fn parse_field(ast: &AST, component: &ComponentInfo) -> Vec<FieldComponent> {
+    let variables: Vec<String> = ast
+        .find_all_children_by_type(&["variable_declarator"])
+        .get_or_insert(vec![])
+        .iter()
+        .map(|var_decl| var_decl.find_child_by_type(&["identifier"]))
+        .filter(|var_ident| var_ident.is_some())
+        .map(|identifier| identifier.unwrap().value.clone())
```

Comment:
I've seen you use it elsewhere in your code, but I think flat_map would be good to use here